py_ecc/field_elements.py

```diff
@@ -40,21 +23,33 @@ def inv(a: int, n: int) -> int:
 # and it becomes a field element.
 class FQ(object):
     n = None  # type: int
+    field_modulus = None
+    curve_name = None
+
+    def __init__(self, val: IntOrFQ, curve_name: str) -> None:
+        """
+        curve_name can be either 'bn128' or 'bls12_381'
+        This is needed to obtain field_modulus, FQ2_MODULUS_COEFFS
+        and FQ12_MODULUS_COEFFS from the curve properties
+        """
+        self.curve_name = curve_name
+        self.field_modulus = field_properties[curve_name]["field_modulus"]
+        # See, it's prime!
+        assert pow(2, self.field_modulus, self.field_modulus) == 2
```

Comment:
Runtime assertions are something we try to avoid because they are easily disabled.  Can you convert this to raise a `ValueError`?